wireguard-broker/src/brokers/mio_client.rs

```diff
@@ -158,10 +160,17 @@ impl BrokerClientIo for MioBrokerClientIo {
                 {
                     let bytes = raw_recv(&self.socket, &mut self.recv_buf[x..y])?;
 
+                    // If we've received nothing so far, and raw_recv came up empty,
```

Comment:
Why was this neccesary?

Reply:
I noticed that Rosenpass would generally panic as soon as it started to exchange with a peer; the first Mio poll that happened would always report that nothing had been received yet, resulting in a state of `RxState::RxSize(0)` and causing the `bail!("Invalid state")` below to trigger. This would bubble up through to the top of the program, causing a panic.